addons/account/models/account_move_line.py

```diff
@@ -3186,6 +3186,7 @@ def _get_installments_data(self, payment_currency=None, payment_date=None):
                     'amount_residual_currency_unsigned': -sign * line.discount_amount_currency,
                     'amount_residual_unsigned': -sign * line.discount_balance,
                     'type': 'early_payment_discount',
+                    'discount_max_date': move.invoice_payment_term_id._get_last_discount_date(move.invoice_date)
```

Comment:
Changed, but I wasn't sure if I needed to defined a `discount_amount_unsigned` variable or not. Also, I'm not sure what sign we want when using bills for example, because for now these new variables `discount_amount_currency` and `discount_amount` are only used for invoices